### ISSUE TITLE
Simulating templates like commercial program - findings and possible bugs

### ISSUE THREAD

**Issue by din14970** [pyxem/diffsims#163]
I've been trying to replicate the behavior of ASTAR for template matching for a bit now. While everything works as intended I can get quite different results in the orientation map. I believe the origin is to be found in the different ways the templates are actually generated; different calculated intensities means different "best match" patterns.

So I did a small (non-exhaustive) reverse engineering exercise, simulating the [001] zone axis pattern of FCC Fe for different conditions in ASTAR and checking the resulting spot intensities. Below is a plot of the spot intensity versus the distance of the spot to the Ewald sphere (I hope I have the units correct because there is no indication of units in the file - but changing them by orders of magnitude did not appreciably influence the conclusion) for different values of excitation error (see legend, again no units). 

![image](https://user-images.githubusercontent.com/25320842/115952697-ca197a00-a4e7-11eb-838d-982ef96adb3b.png)

The point is that the intensity of the spots varies nearly perfectly linearly with the distance from the ewald sphere, and that the excitation error only seems to influence the width of this linear shape function. With a pretty random scaling factor I could get lines of the form `I = 256 - x / (excitation * factor)` to pass through all the points. So this means that each spot in the diffraction pattern has the same "base intensity", reduced linearly by the excitation error.

I vaguely remember someone telling me that ASTAR does not take into consideration atomic scattering factors, or at least the angular dependence of atomic scattering factors, which would be consistent with these results.

These are the images for the same crystal structure using diffsims

Using the linear shape factor:
![image](https://user-images.githubusercontent.com/25320842/115953543-680f4380-a4ec-11eb-9435-be76c9290b8d.png)

Using the Lorentzian shape factor:
![image](https://user-images.githubusercontent.com/25320842/115953532-57f76400-a4ec-11eb-8150-1a07084c98c4.png)

The difference between different excitation errors is (within a certain range) almost negligible. We don't see this linear relationship and the intensity is clearly dominated by the structure factor (which includes atomic scattering). Still I find it odd that there is almost no difference at all and will investigate whether this is a bug somewhere.

The consequence is that in Pyxem spots near the origin dominate in the correlation calculation. Spots further away tend to be more sensitive to changes in orientation, so boosting their intensity to "weigh" more could improve indexation fidelity.

Since we already had a discussion about excitation errors in #160 I think I may have to look through the entire simulation workflow once more; I find the negligible effect of the shape factor and excitation error very strange. I then also propose to introduce the option for simulation the patterns with `scattering_params = None` or equivalent which would ignore atomic scattering altogether and give us the option to simulate ASTAR-like patterns.

**Comment by pc494** [comment]
This is really interesting @din14970, could you share the code you used to generate the two diffsims figures?

**Comment by din14970** [comment]
```python
from diffsims.generators.diffraction_generator import DiffractionGenerator
import diffpy

beam_energy = 200

diff_gen = DiffractionGenerator(accelerating_voltage=beam_energy,
                                shape_factor_model="linear",  # just change here for lorentzian
                               )

structure_aus = diffpy.structure.loadStructure("1534888.cif")
reciprocal_radius = 1.5
dwf = {"Fe": 0.1}

fig, ax = plt.subplots()

for i in [0.01, 0.05, 0.1, 0.5]:
    pattern = diff_gen.calculate_ed_data(
        structure_aus,
        reciprocal_radius,
        with_direct_beam=True,
        max_excitation_error=i,
        debye_waller_factors=dwf,
      )
    condition = pattern.indices[:,2] == 0  # exclude out of plane reflections for simplicity
    x = pattern.coordinates[:,0][condition]
    y = pattern.coordinates[:,1][condition]
    intensity = pattern.intensities[condition]
    
    ewald_r = 1 / (acceleration_voltage_to_wavelength(beam_energy*1000)*1e9)
    z = ewald_r - np.sqrt(ewald_r**2 - x**2 - y**2)
    ax.scatter(z, intensity/intensity.max(), label=i)

ax.set_xlabel("z-height of ewald sphere intersection")
ax.set_ylabel("intensity")


ax.set_ylim(0,0.15)
ax.set_xlim(0,0.0025)

ax.legend()
```

The CIF file (just rename extension, github doesn't allow .cif):
[1534888.txt](https://github.com/pyxem/diffsims/files/6371618/1534888.txt)

**Comment by pc494** [comment]
> The difference between different excitation errors is (within a certain range) almost negligible. We don't see this linear relationship and the intensity is clearly dominated by the structure factor (which includes atomic scattering). Still I find it odd that there is almost no difference at all and will investigate whether this is a bug somewhere.

I had a quick scan of this, using tilts rather than changing excitation error and was reasonably happy with what I saw. I will be interested to see what you turn up.

> The consequence is that in Pyxem spots near the origin dominate in the correlation calculation. Spots further away tend to be more sensitive to changes in orientation, so boosting their intensity to "weigh" more could improve indexation fidelity.

I agree.

> Since we already had a discussion about excitation errors in #160 I think I may have to look through the entire simulation workflow once more; I find the negligible effect of the shape factor and excitation error very strange. I then also propose to introduce the option for simulation the patterns with `scattering_params = None` or equivalent which would ignore atomic scattering altogether and give us the option to simulate ASTAR-like patterns.

I think this would be a good extension to the code as well, as it will give users and extra option they can take or leave as they see fit.

**Comment by din14970** [comment]
This should be resolved by #166
